.gitignore

```diff
@@ -0,0 +1,9 @@
+# Ignore backup files created by editors like emacs.
+*~
+# Ignore cmake files.
+*CMakeCache.txt
+*CMakeFiles
+# Ignore tmp directory/file. We use it sometimes to write debug information.
+tmp
+# Ignore build trees.
+build-*
```

Comment:
The hyphen in `build-` is part of the directory name. Any directory with name starting with `build-` will be ignored. This is to allow having build trees in the directory (which I do all the time). If we later decide to have a directory that starts with `build-` in the repository, we can remove/change this one line.